pkg/sink/sink_test.go

```diff
@@ -478,3 +481,59 @@ func Test_addLabels(t *testing.T) {
 		t.Error(diff)
 	}
 }
+
+type fakeHTTPInterceptor struct {
+	i int32
+}
+
+func (f *fakeHTTPInterceptor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	data := map[string]interface{}{
+		"i": atomic.AddInt32(&f.i, 1),
+	}
+	_ = json.NewEncoder(w).Encode(data)
+}
+
+func TestExecuteInterceptor(t *testing.T) {
```

Comment:
this test is super cool: might be worth adding a docstring explaining how it works? e.g. uses 2 interceptors which increment an int so you can be sure they both are executed, and in order, etc.

Reply:
Done.